pkg/reconciler/managed/reconciler.go

```diff
@@ -662,12 +662,15 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		"external-name", meta.GetExternalName(managed),
 	)
 
-	// If managed resource has a deletion timestamp and and a deletion policy of
+	// If managed resource has a deletion timestamp and a deletion policy of
 	// Orphan, we do not need to observe the external resource before attempting
 	// to unpublish connection details and remove finalizer.
 	if meta.WasDeleted(managed) && managed.GetDeletionPolicy() == xpv1.DeletionOrphan {
 		log = log.WithValues("deletion-timestamp", managed.GetDeletionTimestamp())
-
+		if len(managed.GetFinalizers()) > 1 {
+			log.Debug("Waiting for additional finalizers to be removed", "finalizer_count", len(managed.GetFinalizers()))
+			return reconcile.Result{Requeue: true}, nil
+		}
```

Comment:
I wonder if there's a chance of a deadlock here. Probably quite unlikely?

I'm imagining some scenario in which an external system has added a finalizer that it won't remove until the underlying external resource is gone, the connection secret is gone, etc.

Reply:
Cyclic dependencies are always a possibility but I agree they should be quite unlikely.